sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RetryUtils.java

```diff
@@ -27,16 +27,17 @@
             if (e == null) {
                 return Flux.error(t);
             }
-            policy.captureStartTimeIfNotSet();
+            RetryContext retryContext = policy.getRetryContext();
+            if (retryContext != null) {
+                retryContext.captureStartTimeIfNotSet();
+            }
+
             Flux<ShouldRetryResult> shouldRetryResultFlux = policy.shouldRetry(e).flux();
             return shouldRetryResultFlux.flatMap(s -> {
                 CosmosException clientException = Utils.as(e, CosmosException.class);
-                if(clientException != null) {
-                    policy.addStatusAndSubStatusCode(null, clientException.getStatusCode(), clientException.getSubStatusCode());
-                }
+                    addStatusSubStatusCodeOnRetryContext(retryContext, clientException, s.nonRelatedException);
```

Comment:
Indentation

Reply:
Indentation is correct , will add the brackets and will resolve other comment